influunt-api/modules/influunt-core/app/status/AlarmesFalhasControlador.java

```diff
@@ -77,31 +79,38 @@ public static MongoCollection alarmesFalhas() {
         //TODO: Confirmar se o last nao pega um registro aleatorio. Ele pode ser causa de inconsitencia
         HashMap<String, TipoEvento> hash = new HashMap<>();
         Aggregate.ResultsIterator<Map> ultimo =
-            alarmesFalhas().aggregate("{$sort:{timestamp:-1}}").and("{$group:{_id:'$idControlador', 'timestamp': {$max:'$timestamp'}, 'tipoEvento': {$first: '$conteudo.tipoEvento.tipo'}}}").
-                as(Map.class);
+            alarmesFalhas()
+                .aggregate("{$sort:{timestamp:-1}}")
+                .and("{$group:{_id:'$idControlador', 'timestamp': {$first:'$timestamp'}, 'tipoEvento': {$first: '$conteudo.tipoEvento.tipo'}}}")
+                .as(Map.class);
         for (Map m : ultimo) {
             hash.put(m.get("_id").toString(), TipoEvento.valueOf(m.get("tipoEvento").toString()));
         }
 
         return hash;
     }
 
-    public static List<AlarmesFalhasControlador> ultimosAlarmesFalhasControladores(Integer limit, String query) {
+    public static List<AlarmesFalhasControlador> ultimosAlarmesFalhasControladores(Integer limit, String query, List<String> ids) {
         ArrayList<AlarmesFalhasControlador> list = new ArrayList<>();
 
-        ArrayList<String> predicates = new ArrayList<>();
-        if (query == null) {
-            predicates.add("{ $match: { recuperado: false }  }");
-        } else {
-            predicates.add("{ $match: { recuperado: false, 'conteudo.descricaoEvento': { $regex: '"+ query +"', $options: 'i' } } }");
+        long ultimos30Dias = DateTime.now().minusDays(30).getMillis();
+        String controladoresIds = "[\"" + StringUtils.join(ids, "\",\"") + "\"]";
+
+        Aggregate jongoQuery = alarmesFalhas().aggregate("{ $match: { recuperado: false }  }")
+            .and("{ $match: { idControlador: {$in: " + controladoresIds + "} } }");
+        jongoQuery.and(" { $match: { timestamp: { $gt: " + ultimos30Dias + " } } } ");
+        if (query != null) {
+            jongoQuery.and("{ $match: { 'conteudo.descricaoEvento': { $regex: '"+ query +"', $options: 'i' } } }");
         }
-        predicates.add("{ $sort: { timestamp: -1 } }");
-        predicates.add("{ $project: { _id: 0, idControlador: 1, idAnel: 1, timestamp: 1, conteudo: 1 } }");
+
+        jongoQuery.and("{ $sort: { timestamp: -1 } }");
+        jongoQuery.and("{ $project: { _id: 0, idControlador: 1, idAnel: 1, timestamp: 1, conteudo: 1 } }");
```

Comment:
Não é necessário chamar o método `and` sempre no objeto `jongoQuery`, ele é chainable:
```
jongoQuery.and("...")
  .and("...")
  .and("...")
```